govcd/api_vcd_test.go

```diff
@@ -293,7 +300,38 @@ func (vcd *TestVCD) removeLeftoverEntities(entity CleanupEntity) {
 		}
 		vcd.infoCleanup(removedMsg, entity.EntityType, entity.Name, entity.CreatedBy)
 		return
-
+	case "catalogItem":
+		if entity.Parent == "" {
+			vcd.infoCleanup("removeLeftoverEntries: [ERROR] No Org provided for catalogItem '%s'\n", strings.Split(entity.Parent, "|")[0])
+			return
+		}
+		org, err := GetAdminOrgByName(vcd.client, strings.Split(entity.Parent, "|")[0])
+		if org == (AdminOrg{}) || err != nil {
+			vcd.infoCleanup("removeLeftoverEntries: [INFO] organization '%s' not found\n", entity.Parent)
+			return
+		}
+		catalog, err := org.FindCatalog(strings.Split(entity.Parent, "|")[1])
+		if catalog == (Catalog{}) || err != nil {
+			vcd.infoCleanup(notFoundMsg, entity.EntityType, entity.Name)
+			return
+		}
+		for _, catalogItems := range catalog.Catalog.CatalogItems {
+			for _, catalogItem := range catalogItems.CatalogItem {
+				if catalogItem.Name == entity.Name {
+					catalogItemApi, err := catalog.FindCatalogItem(catalogItem.Name)
+					if catalogItemApi == (CatalogItem{}) || err != nil {
+						vcd.infoCleanup("removeLeftoverEntries: [INFO] catalogItem '%s' not found\n", entity.Name)
+						return
+					}
+					err = catalogItemApi.Delete()
+					if err != nil {
+						vcd.infoCleanup(notDeletedMsg, entity.EntityType, entity.Name, err)
+					}
+				}
+			}
+		}
+		vcd.infoCleanup(removedMsg, entity.EntityType, entity.Name, entity.CreatedBy)
```

Comment:
Something is missing in this procedure. When I use a non existing OVA in the test, it still prints `Removed catalogItem '<ITEM_NAME>' created by Test_UploadOvf`.